src/cr/cube/matrix/measure.py

```diff
@@ -159,6 +161,30 @@ def _subtotal_columns(self):
         # --- arises.
         return SumSubtotals.subtotal_columns(self._base_values, self._dimensions)
 
+    @lazyproperty
+    def _subtotal_rows(self):
+        """2D np.int64 ndarray of column-proportions denominator for subtotal rows.
+
+        This is the third "block" and has the shape (n_row_subtotals, n_cols).
+        """
+        # --- the strategy here is simply to broadcast the columns_base to the shape of
+        # --- the subtotal-rows matrix because a subtotal-row value has the same
+        # --- column-base as all other cells in that column. Note that this initial
+        # --- subtotal-rows matrix is used only for its shape (and when it is empty)
+        # --- because it computes the wrong cell values for this case.
+        subtotal_rows = SumSubtotals.subtotal_rows(self._base_values, self._dimensions)
+
+        # --- in the "no-row-subtotals" case, short-circuit with a (0, ncols) array
+        # --- return value, both because that is the right answer, but also because the
+        # --- non-empty columns-base array cannot be broadcast into that shape. dtype
+        # --- must be `int` to avoid changing type of assembled array to float.
+        if subtotal_rows.shape[0] == 0:
+            return subtotal_rows
+
+        return np.broadcast_to(
```

Comment:
Are we sure this won't broadcast to a wrong shape if the number of rows and columns is the same?

Reply:
Because this is broadcasting "vertically", no. The trailing dimension of `subtotal_rows` will always match the length of the columns-marginal (`.columns_base` I think in this case, the code moved so might be `.columns_margin`, but same applies). So if `subtotal_rows` is (3, 3) and `columns_margin` is (3,), the trailing dimension matches (and always will here). Since that is the only dimension in the margin, the (3,) margin is broadcast to have 3 rows and a shape of (3, 3).

```python
>>> import numpy as np
>>> A = np.arange(9).reshape(3, 3)
>>> A
array([[0, 1, 2],
       [3, 4, 5],
       [6, 7, 8]])
>>> B = np.array([1,2,3])
>>> B
array([1, 2, 3])
>>> np.broadcast_to(B, A.shape)
array([[1, 2, 3],
       [1, 2, 3],
       [1, 2, 3]])
```